rust/src/lib.rs

```diff
@@ -34,54 +48,66 @@ impl GraphWrapper {
         }
     }
 
-    pub fn get_modules(&self) -> FxHashSet<String> {
+    pub fn get_modules(&self) -> HashSet<String> {
         self._graph
-            .get_modules()
-            .iter()
-            .map(|module| module.name.clone())
+            .all_modules()
+            .visible()
+            .names_as_strings()
             .collect()
     }
 
+    pub fn contains_module(&self, name: &str) -> bool {
+        match self.get_visible_module_by_name(name) {
+            Ok(_) => true,
+            Err(GrimpError::ModuleNotPresent(_)) => false,
+            _ => panic!("unexpected error checking for module existence"),
+        }
+    }
+
     #[pyo3(signature = (module, is_squashed = false))]
     pub fn add_module(&mut self, module: &str, is_squashed: bool) -> PyResult<()> {
-        let module_struct = Module::new(module.to_string());
-
-        if let Some(ancestor_squashed_module) =
-            self._graph.find_ancestor_squashed_module(&module_struct)
+        for ancestor_module in self
+            ._graph
+            .module_name_to_self_and_ancestors(module)
+            .into_iter()
+            .skip(1)
```

Comment:
Could you explain this line?

Reply:
The `skip(1)` removes the self, leaving just the ancestors. I'll comment that in the code?

Reply:
Sounds good!